build: macOS toolchain bump

fanquake opened the pull request:
Bumps our macOS toolchain to be using the following:
* Clang 10.0.1 (gitian) & Clang 10.0.0 (Guix)
* ld64 609
* libtapi 1100.0.11
* cctools  973.0.1
* Xcode 12.1
* macOS SDK 10.15.6

which are currently the most recent releases available as open source. See upstream [`cctools`](https://github.com/tpoechtrager/cctools-port/commits/973.0.1-ld64-609) and [`libtapi`](https://github.com/tpoechtrager/apple-libtapi/tree/1100.0.11).

This should improve the possibility of Apple ARM cross-compilation in depends.

This also removes our [patching out of pthreads usage](https://github.com/bitcoin/bitcoin/blob/master/depends/patches/native_cctools/ld64_disable_threading.patch) in `ld64`. There have been multiple changes since `ld64 450.3`, which have likely fixed the non-determinism we were working around. i.e from [InputFiles.cpp](https://opensource.apple.com/source/ld64/ld64-609/src/ld/InputFiles.cpp.auto.html):
```cpp
// <rdar://problem/15002251> make implicit dylib order be deterministic by sorting by install_name
		std::sort(implicitDylibs.begin(), implicitDylibs.end(), DylibByInstallNameSorter());
```

```cpp
// <rdar://problem/42675402> ld64 output is not deterministic due to dylib processing order
		std::sort(unprocessedDylibs.begin(), unprocessedDylibs.end(), [](const ld::dylib::File* lhs, const ld::dylib::File* rhs) {
			return strcmp(lhs->path(), rhs->path()) < 0;
		});
```

Guix Build:
```bash
find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
f6c3817b8fe5f7370299d1ae2533e4a3acd313ba9f9aa8d423a8956117e52dd5  guix-build-a5550f877a2c/output/dist-archive/bitcoin-a5550f877a2c.tar.gz
4954dcf563c2d496b8d9fecd48f8e3f7fba2f319ffa254a5bc8ee12cfee6acf0  guix-build-a5550f877a2c/output/x86_64-apple-darwin18/bitcoin-a5550f877a2c-osx-unsigned.dmg
8f6095b445c7f1a8e6accd86bb7f0696d5849402084927d2b726b7d557831c3a  guix-build-a5550f877a2c/output/x86_64-apple-darwin18/bitcoin-a5550f877a2c-osx-unsigned.tar.gz
cc40f25477b4defc1617ae694313d80f307ddf6742fe6cc85c6bc0e215ef8be0  guix-build-a5550f877a2c/output/x86_64-apple-darwin18/bitcoin-a5550f877a2c-osx64.tar.gz
```

Gitian Build:
```bash
Generating report
506a8abdefe559999b43dd9f14905b9b2b5a3363b1cd013d45ae47acc1f7ef6c  bitcoin-a5550f877a2c-osx-unsigned.dmg
f606997f74026dd12d110d683c6f116b40df324836904ef507dd7ac787e6ebe2  bitcoin-a5550f877a2c-osx-unsigned.tar.gz
5b495ef15f2c3260c2950921b61326912a9bf533cccd51e13818809fd225489e  bitcoin-a5550f877a2c-osx64.tar.gz
f6c3817b8fe5f7370299d1ae2533e4a3acd313ba9f9aa8d423a8956117e52dd5  src/bitcoin-a5550f877a2c.tar.gz
9eb0221e962d2839770963bd03c6c9e98e8bf3078566bee2ae42f06233a710fa  bitcoin-core-osx-22-res.yml
Done.
```

fanquake committed:
build: libtapi 1100.0.11

jonasschnelli commented:
should someone verify if the right sources have been taken from upstream (unchanged)?
https://github.com/tpoechtrager/apple-libtapi
https://github.com/apple/llvm-project/commits/apple/stable/20190104

Why is the snv suffix in the library name required?

fanquake commented:
> Why is the snv suffix in the library name required?

The `.8svn` suffix comes from LLVMs dylib naming, which is the LLVM  major version number + the LLVM version suffix (`svn`). After building you can grep and find:
```bash
build/tools/llvm-config/BuildVariables.inc
35:#define LLVM_DYLIB_VERSION "8svn"

build/include/llvm/Config/config.h
330:#define PACKAGE_STRING "LLVM 8.0.0svn"
333:#define PACKAGE_VERSION "8.0.0svn"

build/include/llvm/Config/llvm-config.h
78:#define LLVM_VERSION_STRING "8.0.0svn"
```

etc.

> should someone verify if the right sources have been taken from upstream (unchanged)?

I have done some diffs and can post them shortly.

fanquake commented:
> should someone verify if the right sources have been taken from upstream (unchanged)?

I've posted a diff of the Clang and libtapi sources here: https://gist.github.com/fanquake/1512109cc69d0a61f352e326f34bb90a.

You can recreate the using the following steps:
Clang:
```bash
git clone https://github.com/apple/llvm-project
git clone https://github.com/tpoechtrager/apple-libtapi

cd llvm-project
git checkout 729748d085a90bd2a4af36efbfb2dc33b4704de3

rm -rf clang/test
rm -rf clang/unittests/
rm -rf clang/www
cd ..

# compare clang
diffoscope --exclude-directory-metadata=yes llvm-project/clang apple-libtapi/src/llvm/projects/clang/
```

libtapi:
```bash
wget https://opensource.apple.com/tarballs/tapi/tapi-1100.0.11.tar.gz
tar xf tapi-1100.0.11.tar.gz
cd tapi-1100.0.11.tar.gz
rm -rf test/
rm -rf unittests/
cd ..

# compare libtapi
diffoscope --exclude-directory-metadata=yes tapi-1100.0.11 apple-libtapi/src/libtapi
```

If anyone wants an intro to `.tbd` stubs, I wrote something here using libconsensus: https://github.com/fanquake/core-review/blob/master/tbd-stubs.md.

promag commented:
> If anyone wants an intro to `.tbd` stubs, I wrote something here using libconsensus: https://github.com/fanquake/core-review/blob/master/tbd-stubs.md.

TIL, very nice! 👏

MarcoFalke commented:
```
make: Entering directory '/home/ubuntu/build/bitcoin/depends'
Extracting native_cctools...
/home/ubuntu/cache/common/55562e4073dea0fbfd0b20e0bf69ffe6390c7f97.tar.gz: OK
/home/ubuntu/cache/common/clang-llvm-8.0.0-x86_64-linux-gnu-ubuntu-14.04.tar.xz: OK
sha256sum: /home/ubuntu/cache/common/86f43cdb62a3ceb39f3ee6e4568eded67a4912e8.tar.gz: No such file or directory
/home/ubuntu/cache/common/86f43cdb62a3ceb39f3ee6e4568eded67a4912e8.tar.gz: FAILED open or read
sha256sum: WARNING: 1 listed file could not be read
funcs.mk:260: recipe for target '/home/ubuntu/build/bitcoin/depends/work/build/x86_64-apple-darwin16/native_cctools/55562e4073dea0fbfd0b20e0bf69ffe6390c7f97-cd4b9b6d54b/.stamp_extracted' failed
make: *** [/home/ubuntu/build/bitcoin/depends/work/build/x86_64-apple-darwin16/native_cctools/55562e4073dea0fbfd0b20e0bf69ffe6390c7f97-cd4b9b6d54b/.stamp_extracted] Error 1
make: Leaving directory '/home/ubuntu/build/bitcoin/depends'

MarcoFalke commented:
Steps to reproduce:

```
   git checkout bitcoin-core/master 
   cd depends/
   make download HOST=x86_64-apple-darwin16 
   git cherry-pick 826d42903da5a927f296be1fa86096bce53d5f1b
   make HOST=x86_64-apple-darwin16

laanwj commented:
> If anyone wants an intro to .tbd stubs, I wrote something here using libconsensus: https://github.com/fanquake/core-review/blob/master/tbd-stubs.md.

I've wanted this kind of thing forever. I wish this was cross-platform.

fanquake commented:
This needs updating for a newer [libtapi update](https://github.com/tpoechtrager/apple-libtapi/commit/a66284251b46d591ee4a0cb4cf561b92a0c138d8), and one of either @dongcarl's or @hebasto's "fix building dependant libs" commits pulled in to fix the rebuilding.

hebasto committed:
build: Clang 10.0.1

LLVM 8 has inherent nondeterminism in the compiler, fixed in LLVM 9+.

jonasschnelli commented:
@MarcoFalke: you need to clear the `depends/sources/` dir (a `make clean`) is not sufficient.

Tested ACK 1a1fc87169bc18db6a9897595401cf81c9b7427a - tested with Xcode-12.2-12B45b

MarcoFalke commented:
depends should ideally detect missing or outdated sources automatically and download them. This is how it used to work, am I wrong?

fanquake commented:
Seeing as this is now useful for a reason other than mentioned in the OP, I'll address outstanding issues and take this out of draft.

> depends should ideally detect missing or outdated sources automatically and download them. This is how it used to work, am I wrong?

IIRC this has never been the case for the `native_cctools` package. However it's always been masked by the fact that we'd always be bumping Clang when changing libtapi, so you'd always be rebuilding anyways.

fanquake committed:
build: native cctools 973.0.1, ld64 609

fanquake committed:
build: Xcode 12.1, macOS SDK 10.15.6

fanquake commented:
I've modified this to be a macOS toolchain bump, based on #21457, rather than updating libtapi individually.

hebasto commented:
un-draft?
:)

fanquake committed:
build: no longer patch threading out of ld64

Changes introduced in ld64-450.3 have likely removed the need for us to
patch out pthreads. See:
https://opensource.apple.com/source/ld64/ld64-450.3/src/ld/InputFiles.cpp.auto.html.

hebasto commented:
Update https://github.com/bitcoin/bitcoin/blob/d9172f83e0bcdf877c25e044a13690bca99bac5a/contrib/gitian-descriptors/gitian-osx.yml#L34 ?

fanquake commented:
> Update
> (everywhere)

Updated the missed replacement in the gitian descriptor, and version numbers in the CI. Was there something else missing?

I've run Guix builds @ d9172f83e0bcdf877c25e044a13690bca99bac5a with various `MAX_JOBS`, (1,3,7,8), and have so far seen deterministic results:
```bash
bash-5.1# find output -type f -name *$(git rev-parse --short HEAD)*.* -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
2862d0d9c21e5508b29932a6bf72a7b3e042e8283f6152ac421631ffa135f962  output/bitcoin-d9172f83e0bc-osx-unsigned.dmg
d8e664c29b2998a6eb66ccf10afa07aa77ca9b11bbcba29c4f7fc2f4e5735810  output/bitcoin-d9172f83e0bc-osx-unsigned.tar.gz
6ddd8eabe273518fdaa7aa0a848d832015d4463ed20d9a6ca8d2a9d411f2554c  output/bitcoin-d9172f83e0bc-osx64.tar.gz
485c65bfd306803363b23a1bc0039f9c060513a1cc8540d113dc06ea71f59209  output/src/bitcoin-d9172f83e0bc.tar.gz
```

hebasto commented:
https://github.com/bitcoin/bitcoin/blob/e4a2918238559e0b82e287833f51b321b97518ab/contrib/gitian-build.py#L213

hebasto commented:
#### Gitian build:
```
Generating report
52301a10a1e988dffa2ac71ece046c5c9de52578ef5f8ff06aa5fc7645a8e694  bitcoin-1e18d58d10d6-osx-unsigned.dmg
4e0174e212ea233552d97299f304f08776263d45a0b1fccabdda60249083d30e  bitcoin-1e18d58d10d6-osx-unsigned.tar.gz
82c36661f208f8501b6310d77b9b30d3b067d05614eb1fb9e575a08bb0a05580  bitcoin-1e18d58d10d6-osx64.tar.gz
97326c1f6df4e48283b33c2fe10be60ba91bd4c5612f9dd97d9a33f2342423f4  src/bitcoin-1e18d58d10d6.tar.gz
03b03010b67b0590d0588b971a86ad5a0e414adfa5435e6a2ba5db9432eba53e  bitcoin-core-osx-22-res.yml
Done.
```

hebasto commented:
#### Guix build:
```
$ find output -type f -name *$(git rev-parse --short HEAD)*.* -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
d965da3230f73a78c91473bf393b8cab6081a539fe335eca573f6bc4e3a4bb9a  output/bitcoin-1e18d58d10d6-osx-unsigned.dmg
03c8b3575a0c6cb402dd8b7f9e1097a3f7303d1c90a562f8e54e863be9b7d9a0  output/bitcoin-1e18d58d10d6-osx-unsigned.tar.gz
e2603c3a55d2b948e46bcc3e481222f081c535806e0311ea9074227bfa755c3d  output/bitcoin-1e18d58d10d6-osx64.tar.gz
97326c1f6df4e48283b33c2fe10be60ba91bd4c5612f9dd97d9a33f2342423f4  output/src/bitcoin-1e18d58d10d6.tar.gz
```

hebasto commented:
Looks like a new sdk archive should be uploaded to https://bitcoincore.org/depends-sources/sdks.

hebasto reviewed (commented):
Approach ACK 1e18d58d10d6021fc1833ea530963e42d2e4c5e6, going to look into the code closer.

fanquake commented:
Guix:
Ran builds with varying `MAX_JOBS=`. i.e:
```bash
time BASE_CACHE="/guix/base_cache" SOURCES_PATH="/guix/sources" SDK_PATH="/guix/SDKs" HOSTS="x86_64-apple-darwin18" MAX_JOBS="1" ./contrib/guix/guix-build.sh
```
all produced the same result.
```bash
bash-5.1# find output -type f -name *$(git rev-parse --short HEAD)*.* -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
d965da3230f73a78c91473bf393b8cab6081a539fe335eca573f6bc4e3a4bb9a  output/bitcoin-1e18d58d10d6-osx-unsigned.dmg
03c8b3575a0c6cb402dd8b7f9e1097a3f7303d1c90a562f8e54e863be9b7d9a0  output/bitcoin-1e18d58d10d6-osx-unsigned.tar.gz
e2603c3a55d2b948e46bcc3e481222f081c535806e0311ea9074227bfa755c3d  output/bitcoin-1e18d58d10d6-osx64.tar.gz
97326c1f6df4e48283b33c2fe10be60ba91bd4c5612f9dd97d9a33f2342423f4  output/src/bitcoin-1e18d58d10d6.tar.gz
```

Gitian:
```bash
52301a10a1e988dffa2ac71ece046c5c9de52578ef5f8ff06aa5fc7645a8e694  bitcoin-1e18d58d10d6-osx-unsigned.dmg
4e0174e212ea233552d97299f304f08776263d45a0b1fccabdda60249083d30e  bitcoin-1e18d58d10d6-osx-unsigned.tar.gz
82c36661f208f8501b6310d77b9b30d3b067d05614eb1fb9e575a08bb0a05580  bitcoin-1e18d58d10d6-osx64.tar.gz
97326c1f6df4e48283b33c2fe10be60ba91bd4c5612f9dd97d9a33f2342423f4  src/bitcoin-1e18d58d10d6.tar.gz
ce3741edbdf340164474a136309ba4ad49ce24bf93e206350429134975b56d41  bitcoin-core-osx-22-res.yml
```

hebasto commented:
Tested 1e18d58d10d6021fc1833ea530963e42d2e4c5e6
On macOS Big Sur 11.2.3 (20D91) the dark mode (https://github.com/bitcoin-core/gui/pull/154) does not work.

goums commented:
ACK 1e18d58d10d6021fc1833ea530963e42d2e4c5e6
Tested on macOS Big Sur 11.2.3.

> Tested [1e18d58](https://github.com/bitcoin/bitcoin/commit/1e18d58d10d6021fc1833ea530963e42d2e4c5e6)
> On macOS Big Sur 11.2.3 (20D91) the dark mode ([bitcoin-core/gui#154](https://github.com/bitcoin-core/gui/pull/154)) does not work.

Dark mode is ok on my side after a clean rebuild and `make deploy`: 
![image](https://user-images.githubusercontent.com/5577626/113170264-afd0e100-9246-11eb-8ae4-fbe572e4f678.png)
Note that I've re ran all build steps from the [build-osx readme ](https://github.com/bitcoin/bitcoin/blob/master/doc/build-osx.md) to avoid any version conflicts.

hebasto commented:
> ACK [1e18d58](https://github.com/bitcoin/bitcoin/commit/1e18d58d10d6021fc1833ea530963e42d2e4c5e6)
> Tested on macOS Big Sur 11.2.3.
> 
> > Tested [1e18d58](https://github.com/bitcoin/bitcoin/commit/1e18d58d10d6021fc1833ea530963e42d2e4c5e6)
> > On macOS Big Sur 11.2.3 (20D91) the dark mode ([bitcoin-core/gui#154](https://github.com/bitcoin-core/gui/pull/154)) does not work.
> 
> Dark mode is ok on my side after a clean rebuild and `make deploy`:

Actually, I tested `bitcoin-1e18d58d10d6-osx-unsigned.dmg` gitian build. And broken dark mode support looks weird.

hebasto commented:
> Tested [1e18d58](https://github.com/bitcoin/bitcoin/commit/1e18d58d10d6021fc1833ea530963e42d2e4c5e6)
> On macOS Big Sur 11.2.3 (20D91) the dark mode ([bitcoin-core/gui#154](https://github.com/bitcoin-core/gui/pull/154)) does not work.

@fanquake 
Could bumping Xcode/SDK be premature?

Qt [docs](https://doc.qt.io/qt-5.12/macos.html) mention only Xcode 10 (10.14 SDK) for Qt 5.12 :man_shrugging:

fanquake commented:
> Could bumping Xcode/SDK be premature?

Just because Xcode 10 is listed on that page, doesn't mean it's the only version that will work (I assume 10 was just the latest Xcode release at the time). We are already building with Xcode 11 (the Xcode version is also basically irrelevant, and Xcode itself is not even present when cross-compiling), and the 10.15 SDK, and anyone building depends on a macOS machine is already building using Xcode 12+ and an 11.x version of the SDK, and that all currently works fine.

hebasto commented:
FWIW, gitian build with dropped "build: Xcode 12.1, macOS SDK 10.15.6" also a dark mode regression.

Assuming that the root of problem lies elsewhere...

hebasto commented:
Here are results of my tests:
1. dark mode works if commit "build: Clang 10.0.1" is dropped
2. tested `clang+llvm-10.0.0-x86_64-linux-gnu-ubuntu-18.04.tar.xz` -- fails
3. tested `clang+llvm-11.0.0-x86_64-linux-gnu-ubuntu-20.04.tar.xz` -- fails

sipa commented:
```
find output -type f -name *$(git rev-parse --short HEAD)*.* -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
d965da3230f73a78c91473bf393b8cab6081a539fe335eca573f6bc4e3a4bb9a  output/bitcoin-1e18d58d10d6-osx-unsigned.dmg
03c8b3575a0c6cb402dd8b7f9e1097a3f7303d1c90a562f8e54e863be9b7d9a0  output/bitcoin-1e18d58d10d6-osx-unsigned.tar.gz
e2603c3a55d2b948e46bcc3e481222f081c535806e0311ea9074227bfa755c3d  output/bitcoin-1e18d58d10d6-osx64.tar.gz
97326c1f6df4e48283b33c2fe10be60ba91bd4c5612f9dd97d9a33f2342423f4  output/src/bitcoin-1e18d58d10d6.tar.gz
```

fanquake committed:
guix: use Clang 10 for the macOS cross compile

jarolrod reviewed (commented):
https://github.com/bitcoin/bitcoin/blob/c6b30ccb2eee5f80f844f79766591f0a1326ce43/contrib/guix/README.md#L170

Should this PR also handle updating this part of the GUIX readme?

GUIX hashes for 9fe45585d6bed47688bc6f670b72f55372082a30

```
find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum

e0ff68338ea8a63d33b2f6a711a2cd5da345c1af144b63784907ee0944a72a01  guix-build-9fe45585d6be/output/dist-archive/bitcoin-9fe45585d6be.tar.gz
980f4a77f1d03ce64f4dcc66728d2ee4c2aa6f816689966363491f38425a59c7  guix-build-9fe45585d6be/output/x86_64-apple-darwin18/bitcoin-9fe45585d6be-osx-unsigned.dmg
35a93860214e9eebe4da9e8ca92d1e699e717bfdec902fbfff620ed0b8c80822  guix-build-9fe45585d6be/output/x86_64-apple-darwin18/bitcoin-9fe45585d6be-osx-unsigned.tar.gz
d7af2053df1db06bd910984dd32b88873342659907362552bd7029ef4e3e9a6b  guix-build-9fe45585d6be/output/x86_64-apple-darwin18/bitcoin-9fe45585d6be-osx64.tar.gz
```

fanquake commented:
> Should this PR also handle updating this part of the GUIX readme?

Done.

jarolrod commented:
Guix hashes for fa11599c7c6bd8a326781728af5b914ce21449c3:

```
find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum

e111a0d3a05e837f6efb6f8f22291ccbdaf6e28f7ceb6259eb42cc56971733df  guix-build-fa11599c7c6b/output/dist-archive/bitcoin-fa11599c7c6b.tar.gz
446cd8b4afed859f7aac4c69964b7ef12d32db2fc0240b01e33142e13b84e918  guix-build-fa11599c7c6b/output/x86_64-apple-darwin18/bitcoin-fa11599c7c6b-osx-unsigned.dmg
e20d5366b4de95aaa05f52cdacddabb516db3a0dec215558a0306833aba3687e  guix-build-fa11599c7c6b/output/x86_64-apple-darwin18/bitcoin-fa11599c7c6b-osx-unsigned.tar.gz
58dc6696b64def767eab3f2c78797e97e09b7b53778828b287a22ac2c9ad29d2  guix-build-fa11599c7c6b/output/x86_64-apple-darwin18/bitcoin-fa11599c7c6b-osx64.tar.gz
```

hebasto commented:
> Tested [1e18d58](https://github.com/bitcoin/bitcoin/commit/1e18d58d10d6021fc1833ea530963e42d2e4c5e6)
> On macOS Big Sur 11.2.3 (20D91) the dark mode ([bitcoin-core/gui#154](https://github.com/bitcoin-core/gui/pull/154)) does not work.

Another round of tests.

- master (e7776e20ed0ddf41d15b3d2df87a92ea6666226c)

1) build depends
2) substitute `depends/x86_64-apple-darwin18/native/bin/clang` and `depends/x86_64-apple-darwin18/native/lib/clang/` with ones compiled with the _new_ toolchain
3) `make deploy`
4) test `Bitcoin-Core.dmg` on macOS 11.2.3

Result: the dark mode does not work.

- master + cherry-picked 64ea3c52a0d94aa5569a891366e1d26101f3aac0

1) build depends
2) substitute `depends/x86_64-apple-darwin18/native/bin/clang` and `depends/x86_64-apple-darwin18/native/lib/clang/` with ones compiled with the _old_ toolchain
3) `make deploy`
4) test `Bitcoin-Core.dmg` on macOS 11.2.3

Result: the dark mode works as expected.

fanquake committed:
build: use -stdlib++-isystem with Clang 10

fanquake commented:
Added a commit to start using `-stdlib++-isystem` now that we are using Clang 10.

hebasto commented:
I think #21793 should go first :)

fanquake commented:
Rebased for #21793. Will add updated gitian and guix build hashes to the PR description.

hebasto reviewed (approved):
ACK a5550f877a2c46d01bb620ae051c0c8ed0fecd0b

#### Guix build:
```
$ find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
f6c3817b8fe5f7370299d1ae2533e4a3acd313ba9f9aa8d423a8956117e52dd5  guix-build-a5550f877a2c/output/dist-archive/bitcoin-a5550f877a2c.tar.gz
4954dcf563c2d496b8d9fecd48f8e3f7fba2f319ffa254a5bc8ee12cfee6acf0  guix-build-a5550f877a2c/output/x86_64-apple-darwin18/bitcoin-a5550f877a2c-osx-unsigned.dmg
8f6095b445c7f1a8e6accd86bb7f0696d5849402084927d2b726b7d557831c3a  guix-build-a5550f877a2c/output/x86_64-apple-darwin18/bitcoin-a5550f877a2c-osx-unsigned.tar.gz
cc40f25477b4defc1617ae694313d80f307ddf6742fe6cc85c6bc0e215ef8be0  guix-build-a5550f877a2c/output/x86_64-apple-darwin18/bitcoin-a5550f877a2c-osx64.tar.gz
```
#### Gitian build:
```
Generating report
506a8abdefe559999b43dd9f14905b9b2b5a3363b1cd013d45ae47acc1f7ef6c  bitcoin-a5550f877a2c-osx-unsigned.dmg
f606997f74026dd12d110d683c6f116b40df324836904ef507dd7ac787e6ebe2  bitcoin-a5550f877a2c-osx-unsigned.tar.gz
5b495ef15f2c3260c2950921b61326912a9bf533cccd51e13818809fd225489e  bitcoin-a5550f877a2c-osx64.tar.gz
f6c3817b8fe5f7370299d1ae2533e4a3acd313ba9f9aa8d423a8956117e52dd5  src/bitcoin-a5550f877a2c.tar.gz
66a6779b4dcf093d64d536a480737278c05d65d4941bfe15ea342e4e05cb2397  bitcoin-core-osx-22-res.yml
Done.
```

`bitcoin-a5550f877a2c-osx-unsigned.dmg` tested on macOS Big Sur 11.3 (20E232) -- the dark mode works as expected.

Sjors commented on depends/hosts/darwin.mk:
```diff
@@ -1,7 +1,7 @@
-XCODE_VERSION=11.3.1
-XCODE_BUILD_ID=11C505
+OSX_SDK_VERSION=10.15.6
+XCODE_VERSION=12.1
```
Do we want to leapfrog to Xcode 12.5 or just leave this? There is a new SDK version 10.15.7 which was a security update, but I don't know if it's relevant to the things we use.

fanquake replied:
Going to leave this as-is for now. May be revisited soon in regards to proper Apple ARM support.

Sjors reviewed (commented):
Having some trouble doing a gitian build, probably unrelated to this PR though...

```sh
./gitian-build-22.py -u https://github.com/fanquake/bitcoin -o m -b -j 10 -m 20000 -d -D -c -n sjors libtapi_1100011
```

(where gitian-build-22.py refers to the latest version of https://github.com/bitcoin/bitcoin/blob/master/contrib/gitian-build.py) 

```
Compiling libtapi_1100011 MacOS
--- Building for focal amd64 ---
Stopping target if it is up
Error response from daemon: No such container: gitian-target
Error: No such container: gitian-target
Making a new image copy
Starting target
Checking if target is upUnable to find image 'base-focal-amd64:latest' locally
docker: Error response from daemon: pull access denied for base-focal-amd64, repository does not exist or may require 'docker login': denied: requested access to the resource is denied.
See 'docker run --help'.
..............................
Error: No such container: gitian-target
```

hebasto commented:
> Having some trouble doing a gitian build, probably unrelated to this PR though...

What if to re-run from the the `--setup` command?

Sjors commented:
ACK a5550f8

I was able to build on macOS 11.3 with Xcode 12.5 as well as cross-compile on Ubuntu. Those binaries seem to behave.

I also did a gitian build:
```
506a8abdefe559999b43dd9f14905b9b2b5a3363b1cd013d45ae47acc1f7ef6c  bitcoin-a5550f877a2c-osx-unsigned.dmg
f606997f74026dd12d110d683c6f116b40df324836904ef507dd7ac787e6ebe2  bitcoin-a5550f877a2c-osx-unsigned.tar.gz
5b495ef15f2c3260c2950921b61326912a9bf533cccd51e13818809fd225489e  bitcoin-a5550f877a2c-osx64.tar.gz
f6c3817b8fe5f7370299d1ae2533e4a3acd313ba9f9aa8d423a8956117e52dd5  src/bitcoin-a5550f877a2c.tar.gz
70433a6f9a40b0a7feb07392810ed0b084763afc9a0c74e3ef09c514b168e0a8  bitcoin-core-osx-22-res.yml
```

Not sure why `bitcoin-core-osx-22-res.yml` doesn't match: https://gist.github.com/Sjors/d7c1f3bff530a48fa7aa1cdf91f7217c

fanquake commented:
Thanks for testing.
> Not sure why bitcoin-core-osx-22-res.yml doesn't match:

It's not expected that `*-res.yml` files match.
